filters/tee/tee.go

```diff
@@ -0,0 +1,172 @@
+package tee
+
+import (
+	"fmt"
+	log "github.com/Sirupsen/logrus"
+	"github.com/zalando/skipper/filters"
+	"io"
+	"net/http"
+	"net/url"
+	"regexp"
+)
+
+const (
+	Name = "Tee"
+)
+
+type teeSpec struct{}
+
+type teeType int
+
+const (
+	asBackend teeType = iota + 1
+	pathModified
+)
+
+type tee struct {
+	client      *http.Client
+	typ         teeType
+	host        string
+	scheme      string
+	rx          *regexp.Regexp
+	replacement string
+}
+
+type teeTie struct {
+	r io.Reader
+	w *io.PipeWriter
+}
+
+// Returns a new tee filter Spec, whose instances execute
+// the exact same Request against a shadow backend.
+// parameters: shadow backend url, optional - the path(as a regexp) to match and the replacement string.
+// Name: "tee".
+// Example
+// Path("/api/v3") -> tee("https://api.example.com") -> "http://example.org/"
+// This route wil send incoming  request to http://example.org/api/v3 but will also send
+// a copy of the query to https://api.example.com/api/v3.
+// Example
+// Path("/api/v3") -> tee("https://api.example.com", ".*", "/v1/" ) -> "http://example.org/"
+// This route wil send incoming request to http://example.org/api/v3 but will also send
+// a copy of the request to https://api.example.com/v1/ . Note that scheme and path are changed
+
```

Comment:
this empty line makes the above documentation not visible in godoc. Please remove it and then it will be generated automatically like e.g. https://godoc.org/github.com/zalando/skipper/filters/diag#NewBackendBandwidth